web/src/components/FilingsCalendar.test.tsx

```diff
@@ -649,4 +649,58 @@ describe("<FilingsCalendar />", () => {
       expect(screen.queryByText(Config.dashboardDefaults.calendarGridViewButton)).not.toBeInTheDocument();
     });
   });
+
+  describe("calendar list view, view more functionality", () => {
+    function renderCalendarWithEntries(numberOfCalendarEntries: number) {
+      let dueDate: dayjs.Dayjs;
+      let annualReport: TaxFiling;
+      const operateReferences: Record<string, OperateReference> = {};
+      const filings: TaxFiling[] = [];
+
+      for (let i = 0; i < numberOfCalendarEntries; i++) {
+        dueDate = getCurrentDate().add(2, "months").add(i, "day");
+        annualReport = generateTaxFiling({
+          identifier: `annual-report-${i}`,
+          dueDate: dueDate.format(defaultDateFormat),
+        });
+        filings.push(annualReport);
+        operateReferences[`annual-report-${i}`] = generateOperateReference({});
+      }
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => {
+              return obj.displayCalendarType === "FULL" && obj.displayCalendarToggleButton;
+            })
+          ).id,
+        }),
+        taxFilingData: generateTaxFilingData({ filings: filings }),
+        preferences: generatePreferences({ isCalendarFullView: false }),
+      });
+
+      renderFilingsCalendar(operateReferences, userData);
+    }
+
+    it("does not show the view more button if 5 or less events are in the calendar", () => {
+      renderCalendarWithEntries(3);
```

Comment:
with a test like this, i'd usually test the edge of the edge case (ie, 5 filings here instead of 3)

same with the one below on the other side of the edge - I'd use 6 filings instead of 7. This forces the tests to confirm that yes, it is in fact 5 and nothing else. I think these tests would still pass if the numberToShow was set to 4 or 6 in the code, and we don't want that